### PR TITLE
Support React 16

### PR DESCRIPTION
**Pull request by redonkulus**
@lingyan @snyamathi @mridgway 

See https://reactjs.org/blog/2017/09/26/react-v16.0.html

Upgrades:
- fluxible-addons-react
- fluxible-router
- generator-fluxible
- site
- tests

### CODE REVIEW

File: site/tests/unit/actions/demoException.js [hunk 1]
```diff
@@ -18,10 +18,9 @@ describe('site', () => {
 
         it('should execute the action', function (done) {
             demoException(context, {}, function (err) {
-                expect(err).to.eql(new Error('Whoops!'));
+                expect(err).to.eql(new Error('WWhoops!'));
```
Comment (mridgway) [inline site/tests/unit/actions/demoException.js]: Is this legit?

Comment (redonkulus) [reply site/tests/unit/actions/demoException.js]: Doh that was a test ;) fixing...

**Comment by alexparish** [comment]
Great to see!

Could this and https://github.com/yahoo/fluxible/pull/565 be rolled into major releases for all packages?

**Comment by redonkulus** [comment]
@alexparish I will keep them separate as the fluxible router change requires more scrutiny and shouldn't block this PR from being released. 

Plus, I do not plan to bump major for this PR as there should not be any breaking changes. Bumping major is done more carefully as that requires us to update a lot of packages internally.

**Comment by redonkulus** [comment]
Released in fluxible-router@1.4.0 and fluxible-addons-react@0.2.14

**Pull request merged**